core/src/main/kotlin/org/evomaster/core/search/gene/sql/SqlMultiRangeGene.kt

```diff
@@ -0,0 +1,99 @@
+package org.evomaster.core.search.gene.sql
+
+import org.evomaster.core.logging.LoggingUtil
+import org.evomaster.core.output.OutputFormat
+import org.evomaster.core.search.gene.*
+import org.evomaster.core.search.gene.sql.SqlStrings.removeEnclosedQuotationMarks
+import org.evomaster.core.search.service.AdaptiveParameterControl
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.mutator.genemutation.AdditionalGeneMutationInfo
+import org.evomaster.core.search.service.mutator.genemutation.SubsetGeneSelectionStrategy
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class SqlMultiRangeGene<T>(
+        name: String,
+        val template: SqlRangeGene<T>,
+        val rangeGenes: ArrayGene<SqlRangeGene<T>> = ArrayGene(name, template)
+) : Gene(name, mutableListOf(rangeGenes)) where T : ComparableGene {
+
+    companion object {
+        val log: Logger = LoggerFactory.getLogger(SqlMultiRangeGene::class.java)
+    }
+
+    override fun getChildren(): MutableList<Gene> = mutableListOf(rangeGenes)
+
+    override fun copyContent(): Gene {
+        val copyOfRangeGenes = rangeGenes.copyContent() as ArrayGene<SqlRangeGene<T>>
+        return SqlMultiRangeGene(
+                name,
+                template = copyOfRangeGenes.template,
+                rangeGenes = copyOfRangeGenes
+        )
+    }
+
+    override fun randomize(randomness: Randomness, forceNewValue: Boolean, allGenes: List<Gene>) {
+        rangeGenes.randomize(randomness, forceNewValue, allGenes)
+    }
+
+    override fun candidatesInternalGenes(
+            randomness: Randomness,
+            apc: AdaptiveParameterControl,
+            allGenes: List<Gene>,
+            selectionStrategy: SubsetGeneSelectionStrategy,
+            enableAdaptiveGeneMutation: Boolean,
+            additionalGeneMutationInfo: AdditionalGeneMutationInfo?
+    ): List<Gene> {
+        return listOf(rangeGenes)
+    }
+
+    override fun getValueAsRawString(): String {
```

Comment:
for now it is fine. but i guess at certain point might want to refactor `Sql*Gene`  classes to be re-usable in other contexts besides SQL (with maybe just special mode for when they are printed out)